pkg/gather/service/analyze.go

```diff
@@ -0,0 +1,133 @@
+package service
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// AnalyzeGatherBundle will analyze the bootstrap gather bundle at the specified path.
+// Analysis will be logged.
+// Returns an error if there was a problem reading the bundle.
+func AnalyzeGatherBundle(bundlePath string) error {
+	// open the bundle file for reading
+	bundleFile, err := os.Open(bundlePath)
+	if err != nil {
+		return errors.Wrap(err, "could not open the gather bundle")
+	}
+	defer bundleFile.Close()
+	return analyzeGatherBundle(bundleFile)
+}
+
+func analyzeGatherBundle(bundleFile io.Reader) error {
+	// decompress the bundle
+	uncompressedStream, err := gzip.NewReader(bundleFile)
+	if err != nil {
+		return errors.Wrap(err, "could not decompress the gather bundle")
+	}
+	defer uncompressedStream.Close()
+
+	// read through the tar for relevant files
+	tarReader := tar.NewReader(uncompressedStream)
+	var releaseImageAnalysis *analysis
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return errors.Wrap(err, "encountered an error reading from the gather bundle")
+		}
+		if header.Typeflag != tar.TypeReg {
+			continue
+		}
+		filenameParts := strings.SplitN(header.Name, "/", 2)
+		if len(filenameParts) != 2 {
+			continue
+		}
+		// we only care about the release-image.service for now. in the future, we will look at other services, too.
+		if filenameParts[1] == "bootstrap/services/release-image.json" {
+			var err error
+			releaseImageAnalysis, err = analyzeService(tarReader)
+			if err != nil {
+				logrus.Infof("Could not analyze the release-image.service: %v", err)
+			}
+			break
+		}
+	}
+
+	// log details about the release-image.service.
+	if releaseImageAnalysis != nil && releaseImageAnalysis.starts > 0 {
+		if !releaseImageAnalysis.successful {
+			logrus.Error("The bootstrap machine failed to download the release image")
+			for _, l := range strings.Split(releaseImageAnalysis.lastError, "\n") {
+				logrus.Info(l)
+			}
+		}
+	} else {
+		logrus.Error("The bootstrap machine did not execute the release-image.service systemd unit")
+	}
```

Comment:
Despite the comment and recognition that we are limiting the scope right now to analyzing only a single service, I think this is too coupled to release image analysis and it would be a good idea to establish the pattern for future analysis.

I suggest we create a map:
```go
var analysisFuncs map[string]func(io.Reader)
```
where the key is the filename and the value is a func that accepts the file, calls `analyzeService` and handles the result.

Reply:
I kind of like this recommendation, but I'm also concerned about it biasing towards "generate many recommendations".  Sometimes generating many recommendations might be useful, but it also feels like most cases we want to try to boil this down to a single, most-important recommendation.  I dunno how to structure that, vs. the possibly-random tar ordering, so I'm fine punting to follow-up work.